.changelog/unreleased/features/3634-query-app-for-lanes-info.md

```diff
@@ -0,0 +1,3 @@
+- `[mempool/node]` retrieve `LanesInfo` from application and pass on to mempool constructor. 
+Move the ABCI `Info` call from the `Handshake` function to the `NewNodeWithCliParams` function.
+  ([#3634](https://github.com/cometbft/cometbft/pull/3634))
```

Comment:
Question... rather than adding a log line for every incremental change we do on mempool lanes, wouldn't it be better to just add one log at the end (when we merge back to `main`) saying "mempool lanes implemented as per ADR 118".
To me, log lines are different from "git log" in that it is more coarse-grained?
Anyway, this is just a thought... WDYT?

Reply:
I agree with @sergio-mena on having just one log entry for Lanes, but this PR affects other parts of the code (node and consensus). In this case, I would leave a log entry like: 
> `[node]` Move ABCI `Info` call from the `Handshake` function in consensus module to `NewNodeWithCliParams` in node.

Reply:
I don't mind, but I think we should keep note at least in an external doc on the particular changes like this where we break or change APIs. These functions might get lost once we submit a big PR

Reply:
Just a reminder to add a breaking-change changelog file for `Handshake` if its signature changes.

Reply:
@sergio-mena , @hvanz  to be clear, the consensus is we break the API of `Handshake`  ? If not , then I will move 
```
if res == nil {
		var err error
		res, err = proxyApp.Query().Info(ctx, proxy.InfoRequest)
```
into `Handshake` and pass res to ` HandshakeWithABCIRes` instead of `nil`  but remove the repeated query execution from `HandshakeWithABCIRes` as suggested below.

Reply:
Yes, I understood that, I just want to make sure we are on the same page with regards to breaking the API. Regardless of that, the repeated call is going out of the new function.